contrib/node/tests/python/pants_test/contrib/node/tasks/test_node_resolve_src_deps.py

```diff
@@ -0,0 +1,361 @@
+
+  def _create_basic_app(self, src_root=None):
+    src_root = src_root or 'src/node'
+    self.create_file(os.path.join(src_root, 'app/yarn.lock'))
+    self.create_file(os.path.join(src_root, 'app/package.json'), contents=dedent("""
+      {
+        "name": "app",
+        "version": "0.0.1",
+        "bin": "index.js"
+      }
+    """))
+    self.create_file(os.path.join(src_root, 'app/index.js'), contents=dedent("""
+      console.log('hello');
+    """))
+    app = self.make_target(spec=os.path.join(src_root, 'app:app'),
+                           target_type=NodeModule,
+                           sources=['index.js', 'package.json', 'yarn.lock'],
+                           package_name='app',
+                           package_manager='yarn',
+                           bin_executables='index.js')
+    return app
+
+  def _create_workspace(self, with_trans_dep=None):
+    self._create_dep(provide_bin=True, src_root='src/node/workspace/projects', make_target=False)
+    if with_trans_dep:
+      trans_dep = self._create_trans_dep()
+      dep = self._create_dep(trans_dep=trans_dep, provide_bin=True, use_bin_dict=True)
+      app = self._create_app(dep)
+    else:
+      app = self._create_basic_app()
+
+    self.create_file('src/node/workspace/yarn.lock')
+
+    # Workspaces can only be enabled in private projects. 
+    self.create_file('src/node/workspace/package.json', contents=dedent("""
+      {
+        "name": "workspace",
+        "version": "1.0.0",
+        "private": true,
+        "workspaces": [
+          "./projects/dep"
+        ],
+        "dependencies": {
+          "app": "file:../app"
+        }
+      }
+    """))
+
+    workspace = self.make_target(spec='src/node/workspace:workspace',
+                           target_type=NodeModule,
+                           sources=['package.json', 'yarn.lock',
+                                    'projects/**/*'],
+                           package_name='workspace',
+                           package_manager='yarn',
+                           dependencies=[app])
+    return workspace
+
+  def _resolve_target(self, target):
+    context = self.context(target_roots=[target], options={
+      'npm-resolver': {'install_optional': False,
+                       'force_option_override': False,
+                       'install_production': False,
+                       'force': False,
+                       'frozen_lockfile': True}
+    })
+    self.wrap_context(context, [NodePaths])
+    task = self.create_task(context)
+    task.execute()
+
+    node_paths = context.products.get_data(NodePaths)
+    return task, node_paths
+
+  def test_resolve_simple_dep_graph(self):
+    dep = self._create_dep(provide_bin=True)
+    app = self._create_app(dep)
+    _, node_paths = self._resolve_target(app)
+ 
+    dep_node_path = node_paths.node_path(dep)
+    app_node_path = node_paths.node_path(app)
+    self.assertIsNotNone(dep_node_path)
+    self.assertIsNotNone(app_node_path)
```

Comment:
Could you have a stronger assertion here than just isNotNone?

Reply:
There are stronger assertions down the line to verify the paths are correct